openant/devices/shift.py

```diff
@@ -99,10 +140,60 @@ def on_data(self, data):
             self.data["shift"].shift_failure_front = (data[7] & 0xF0) >> 4
 
             delta_update_count = (
-                self._event_count[1] + 256 - self._event_count[0]
+                self._event_count[0][1] + 256 - self._event_count[0][0]
             ) % 256
 
             # if it's a new event (count change)
             if delta_update_count:
                 _logger.info(f"Shifting status update {self}: {self.data['shift']}")
                 self.on_device_data(page, "shift_system_status", self.data["shift"])
+        elif page == 0x02:
+            self._event_count[1][0] = self._event_count[1][1]
+            self._event_count[1][1] = data[1]
+
+            self.data["shift"].event_count = data[1]
+
+            self.data["shift"].event_1 = FunctionSetEvent(data[2] & 0x0F, (data[2] & 0x03) >> 4)
+            self.data["shift"].event_2 = FunctionSetEvent(data[3] & 0x0F, (data[3] & 0x03) >> 4)
+            self.data["shift"].event_3 = FunctionSetEvent(data[4] & 0x0F, (data[4] & 0x03) >> 4)
+            self.data["shift"].event_4 = FunctionSetEvent(data[5] & 0x0F, (data[5] & 0x03) >> 4)
+            self.data["shift"].event_5 = FunctionSetEvent(data[6] & 0x0F, (data[6] & 0x03) >> 4)
+
+            delta_update_count = (
+                self._event_count[1][1] + 256 - self._event_count[1][0]
+            ) % 256
+
+            # if it's a new event (count change)
+            if delta_update_count:
+                _logger.info(f"Shifting status update {self}: {self.data['shift']}")
+                self.on_device_data(page, "shift_system_status", self.data["shift"])
+        elif page == 0x03:
+            self.data["shift"].function_set_1_configuration = FunctionSetConfiguration(data[1] & 0x02, data[1] & 0x04, data[1] & 0x08)
```

Comment:
Same as above, these will be `int` not `bool`. They need casting to `bool` with a operator like `data[1] & 0x02 == 0x02`.